File: csharp/MudbaseShowcase.Events/Services/Rbac.cs
namespace MudbaseShowcase.Events.Services;

/// <summary>
/// Client-/server-side mirror of the RBAC matrix Mudbase's own collection permissions already
/// enforce server-side on the real project (see plan/build-plan.md). Used both to hide/disable
/// controls a role cannot use in the Razor views, and as a second, server-side check inside every
/// mutating Razor Page handler before ever calling Mudbase - it is NOT the real security boundary
/// by itself. A raw API call bypassing this app's UI entirely is still rejected by the platform;
/// see the reference app's plan/build-plan.md "Live Smoke Test Results" (../../web/plan/build-plan.md),
/// carried forward here since both ports share the same live project/collections/roles.
///
/// Booking creation is deliberately NOT gated here: per the reference app's RBAC matrix, an
/// organizer may book another organizer's event (only their own event's Book affordance is
/// hidden, an ownership check, not a role check) - see Pages/Events/Detail.cshtml.cs.
/// </summary>
public static class Rbac
{
    public const string Organizer = "organizer";
    public const string Attendee = "attendee";

    public static bool IsOrganizer(string? role) => role == Organizer;

    public static bool IsAttendee(string? role) => role == Attendee;

    public static string RoleLabel(string? role) => role switch
    {
        Organizer => "Organizer",
        Attendee => "Attendee",
        _ => "Unknown",
    };
}

File: php/src/Support/Rbac.php
<?php

declare(strict_types=1);

namespace App\Support;

/**
 * Client-side mirror of the RBAC matrix Mudbase's own collection permissions already enforce
 * server-side (see plan/build-plan.md). Used purely to hide/disable controls a role cannot use
 * and to show a clear reason why — it is NOT the security boundary. A raw request bypassing this
 * app's views is still rejected by the platform; see the smoke test in build-plan.md. A PHP port
 * of the reference app's `useAuth()` role booleans (src/hooks/useAuth.ts).
 */
final class Rbac
{
    public static function isOrganizer(?string $role): bool
    {
        return $role === 'organizer';
    }

    public static function isAttendee(?string $role): bool
    {
        return $role === 'attendee';
    }

    /**
     * UX-only ownership gate for edit/delete/check-in affordances — organizers see these controls
     * only on events they organize themselves (see plan/build-plan.md "RBAC Matrix"). Mudbase's
     * own collection permissions are the real boundary; a raw request with an organizer token
     * against another organizer's event still succeeds or fails per the platform's actual grant,
     * independent of this check.
     */
    public static function canManageOwnEvent(?string $role, string $organizerId, ?string $userId): bool
    {
        return self::isOrganizer($role) && $userId !== null && $userId !== '' && $organizerId === $userId;
    }

    public static function roleLabel(?string $role): string
    {
        return match ($role) {
            'organizer' => 'Organizer',
            'attendee' => 'Attendee',
            default => 'Unknown',
        };
    }
}
